hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
     """
     Formally publishes a resource in HydroShare. Triggers the creation of a DOI for the resource, and triggers the
     exposure of the resource to the HydroShare DataONE Member Node. The user must be an owner of a resource or an
     adminstrator to perform this action.
 
-    REST URL:  PUT /publishResource/{pid}
+    REST URL:  PUT /resource/{pid}
```

Comment:
@selimnairb I have not done anything to expose a REST end point for publishResource. Not sure whether this comment line was changed in the first place. Should I revert this comment line back? Should I do anything to expose such a REST end point for publishResource in this feature branch or should we do it later as a separate issue tackled by you? Just let me know what is the best way we should go with to handle REST API for resource publication.